pkg/validations/property/anyof.go

```diff
@@ -0,0 +1,175 @@
+	}
+
+	switch in.AdditionPolicy {
+	case AdditionPolicyAllow, AdditionPolicyDisallow:
+		// valid
+	case AdditionPolicy(""):
+		// default to disallow
+		in.AdditionPolicy = AdditionPolicyDisallow
+	default:
+		return fmt.Errorf("%w: %q", errUnknownAdditionPolicy, in.AdditionPolicy)
+	}
+
+	return nil
+}
+
+// AnyOfConfig contains additional configurations for the AnyOf validation.
+type AnyOfConfig struct {
+	// additionPolicy is how adding enums to an existing set of
+	// enums should be treated.
+	// Allowed values are Allow and Disallow.
+	// When set to Allow, adding new values to an existing set
+	// of enums will not be flagged.
+	// When set to Disallow, adding new values to an existing
+	// set of enums will be flagged.
+	// Defaults to Disallow.
+	AdditionPolicy AdditionPolicy `json:"additionPolicy,omitempty"`
+	// Subschemas is a list of subschemas that are part of the anyOf constraint.
+	Subschemas []apiextensionsv1.JSONSchemaProps `json:"subschemas,omitempty"`
+}
+
+// AnyOf is a validation implementation for the "anyOf" constraint in JSONSchemaProps.
+type AnyOf struct {
+	// AnyOfConfig is the set of additional configuration options
+	AnyOfConfig
+	enforcement config.EnforcementPolicy
+}
+
+// Name NewAnyOf creates a new AnyOf validation instance with the provided configuration.
+func (a *AnyOf) Name() string {
+	return anyOfValidationName
+}
+
+// SetEnforcement sets the EnforcementPolicy for the Required validation.
+func (a *AnyOf) SetEnforcement(policy config.EnforcementPolicy) {
+	a.enforcement = policy
+}
+
+// Compare compares an old and a new JSONSchemaProps, checking for incompatible changes to the format constraints of a property.
+// In order for callers to determine if diffs to a JSONSchemaProps have been handled by this validation
+// the JSONSchemaProps.format field will be reset to '""' as part of this method.
+// It is highly recommended that only copies of the JSONSchemaProps to compare are provided to this method
+// to prevent unintentional modifications.
+func (a *AnyOf) Compare(old, newProp *apiextensionsv1.JSONSchemaProps) validations.ComparisonResult {
+	oldList := old.AnyOf
+	newList := newProp.AnyOf
+
+	var err error
+
+	switch {
+	case len(oldList) == 0 && len(newList) > 0:
+		err = fmt.Errorf("%w : net-new anyOf constraint added", ErrNetNewAnyOfConstraint)
+	case len(oldList) > 0 && len(newList) == 0:
+		err = fmt.Errorf("%w : anyOf constraint removed", ErrRemovedAnyOfConstraint)
+	case !anyOfEqual(oldList, newList):
+		err = fmt.Errorf("%w : anyOf subschemas changed", ErrChangedAnyOfConstraint)
+	}
+
+	old.AnyOf = nil
+	newProp.AnyOf = nil
+
+	return validations.HandleErrors(a.Name(), a.enforcement, err)
+}
+
+// anyOfEqual compares two slices of JSONSchemaProps for deep equality.
+func anyOfEqual(a, b []apiextensionsv1.JSONSchemaProps) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for i := range a {
+		if !equalJSONSchemaProps(&a[i], &b[i]) {
+			return false
+		}
+	}
```

Comment:
What if the ordering of `a` is not the same ordering as `b`?

Is there a way for us to, reliably, uniquely identify an `apiextensionsv1.JSONSchemaProps` entry so we only compare like-to-like instance?